functions/Grant-YouTubeOauth.ps1

```diff
@@ -17,23 +17,28 @@ function Grant-YoutubeOauth {
     Start-PodeServer -ScriptBlock {
       Add-PodeEndpoint -Port 8000 -Protocol Http
       Add-PodeRoute -Method Get -Path /auth/complete -ScriptBlock {
-        @{
-          access_token = $WebEvent.Query['access_token']
-        } | ConvertTo-Json | Set-Content -Path $HOME/.pwsh.youtube.oauth.json
-
         $Response = @'
         <h1 style="font-family: sans-serif;">Authentication Complete</h1>
         <h3 style="font-family: sans-serif;">You may close this browser window.</h3>
         <script>
           console.log(window.location.hash);
           let access_token_regex = /access_token=(?<access_token>.*?)&token_type/;
           let result = access_token_regex.exec(window.location.hash);
-          fetch(`/auth/complete?access_token=${result.groups.access_token}`);
+          fetch(`/auth/receive?access_token=${result.groups.access_token}`);
```

Comment:
What's happening here, with the newly added `/receive` route?

Reply:
Check it out a line 33.

/complete is your solution to parse the token. It was invoking itself after parsing
now it invokes /receive
/receive returns the token and closes the pode server. This allows the Powershell job to complete reliably.